File: app/src/main/java/com/martynaskairys/walltip/shared/tracking/UserTracker.java
package com.martynaskairys.walltip.shared.tracking;

import com.martynaskairys.walltip.features.selectcategory.MainActivity;

/**
 * Tracks user activity
 */
public interface UserTracker {

	/** Reports to tracking system that use has entered {@link MainActivity} */
	void reportInChoosingFolderActivityOnCreate();

	void reportInChoosingFolderActivityFolderA();

	void reportInChoosingFolderActivityFolderB();

	void reportInChoosingFolderActivityFolderC();

	void reportInPictureActivityOnCreate();

	void reportInThumbnailActivityOnCreate();
}

File: app/src/main/java/com/martynaskairys/walltip/shared/tracking/UserTracker.kt
package com.martynaskairys.walltip.shared.tracking

import com.martynaskairys.walltip.features.selectcategory.MainActivity

/**
 * Tracks user activity
 */
interface UserTracker {

    /** Reports to tracking system that use has entered [MainActivity]  */
    fun reportInChoosingFolderActivityOnCreate()

    fun reportInChoosingFolderActivityFolderA()

    fun reportInChoosingFolderActivityFolderB()

    fun reportInChoosingFolderActivityFolderC()

    fun reportInPictureActivityOnCreate()

    fun reportInThumbnailActivityOnCreate()
}
